web/src/main/java/org/mskcc/cbio/portal/web/ProxySessionServiceController.java

```diff
@@ -250,12 +250,14 @@ public void updateUsersInVirtualStudy(@PathVariable SessionType type, @PathVaria
 		if (isSessionServiceEnabled() && !(authentication == null
 				|| (authentication instanceof AnonymousAuthenticationToken))) {
 
-			Map<String, Object> map = new HashMap<>();
-			map.put("data.users", ((UserDetails) authentication.getPrincipal()).getUsername());
-			map.put("data.origin", studyIds);
+			String username =  ((UserDetails) authentication.getPrincipal()).getUsername();
 
 			ObjectMapper mapper = new ObjectMapper();
-			String query = mapper.writeValueAsString(map);
+
+			//ignore origin studies order
+            String query = "{ $and: [{ \"data.users\": \"" + username + "\" }, { \"data.origin\": { $all: "
```

Comment:
@kalletlak could you elaborate why the hashmap doesn't work? any clue? Just curious

Reply:
its not about hashmap. new query includes `$and` `$all` etc. which is hard to construct using hashmap

Reply:
+1

Reply:
interesting!